templates/netdev.epp

```diff
@@ -0,0 +1,44 @@
+<%- | String $interface,
+      Integer $dport,
+      Array $peers,
+      Optional[String] $description,
+      Optional[Integer] $mtu,
+| -%>
+# THIS FILE IS MANAGED BY PUPPET
+# based on https://dn42.dev/howto/wireguard
+[NetDev]
+Name=<%= $interface %>
+Kind=wireguard
+<% if $description { -%>
+Description=<%= $description %>
+<% } -%>
+<% if $mtu { -%>
+MTUBytes=<%= $mtu %>
+<% } -%>
+
+[WireGuard]
+PrivateKeyFile=/etc/wireguard/<%= $interface %>
+ListenPort=<%= $dport %>
+
+<% $peers.each |$peer| { -%>
+[WireGuardPeer]
+PublicKey=<%= $peer['public_key'] %>
+<% if $peer['endpoint'] { -%>
+Endpoint=<%= $peer['endpoint'] %>
+<%} -%>
+<% if $peer['persistent_keepalive'] { -%>
+PersistentKeepalive=<%= $peer['persistent_keepalive'] %>
+<% } else { -%>
+PersistentKeepalive=0
+<%} -%>
+<% if $peer['allowed_ips'] { -%>
+<% $peer['allowed_ips'].each |$allowed_ip| { -%>
+AllowedIPs=<%= $allowed_ip %>
+<% } -%>
+<% } else { -%>
+AllowedIPs=fe80::/64
+AllowedIPs=fd00::/8
+AllowedIPs=0.0.0.0/0
```

Comment:
This looks weird. Only allow IPv6 link local but all IPv4 traffic?

Reply:
i don't want to break current logic, so i copied this section from main branch

Reply:
I wanted to make this configureable but had no time yet

Reply:
now its configurable :D should i remove this block?

Reply:
I think it's important to stay backwards compativle for now, so I would like to keep it